### ISSUE TITLE
Questions on Fine-tuning Setup and Data Integration for Time-MoE

### ISSUE THREAD

**Issue by XUHIT** [Time-MoE/Time-MoE#82]
Dear Dr. Shi,

First of all, I would like to sincerely thank you and your team for your excellent work on Time-MoE. The model design is both elegant and effective, and I am especially impressed by the stability and reproducibility of its zero-shot performance across multiple datasets. This consistency is extremely valuable to the research community.
I am currently conducting fine-tuning experiments based on your paper and open-source implementation, and I hope to seek your clarification on a few technical aspects:

1. Integration of Evaluation Datasets
According to the paper, my understanding is that the training splits of the six evaluation datasets (ETTh1, ETTh2, ETTm1, ETTm2, Weather, and GlobalTemp-1000) were merged into a single .jsonl file for fine-tuning.
Based on this, the variable composition would be:

ETT family: 4 datasets × 7 variables = 28 variables
Weather: 21 variables
GlobalTemp-1000: 1000 variables
→ Total ≈ 1049 variables
Could you kindly confirm whether this integration strategy and variable count are correct?

2. Windowing Strategy During Fine-tuning
I noticed that if I use stride=None (default non-overlapping windowing) during fine-tuning, the number of samples becomes extremely limited, especially for shorter sequences.
Under this setting, the performance degrades significantly on certain datasets compared to the zero-shot baseline, which is unexpected.
I am wondering:
Is overlapping windowing (e.g., setting stride < window length) recommended during fine-tuning?
What windowing configuration was used in the paper’s fine-tuning setup?

3. Reproducibility and Experimental Command
For your reference, this is the command I used to perform the fine-tuning:

python torch_dist_run.py main.py \
  -d train_data/all_train.jsonl \
  --precision bf16 \
  --learning_rate 1e-4 \
  --min_learning_rate 5e-5 \
  --lr_scheduler_type cosine \
  --attn_implementation flash_attention_2 \
  --model pretrained_model/TimeMoE-50M \
  --o logs/time_moe_50m-1 \
  --max_length 4096 \
  --global_batch_size 128 \
  --normalization none \
  --micro_batch_size 8 \
  --num_train_epochs 1.0
The experiment was conducted on a single NVIDIA RTX 4090 GPU.

Additionally, I’ve attached a result figure comparing the zero-shot vs. fine-tuned performance. I would greatly appreciate it if you could take a brief look and let me know whether the overall setup—especially the windowing strategy—needs adjustment.

Thank you again for your outstanding contribution and for making your work publicly available. I sincerely appreciate your time and guidance.

<img width="519" alt="Image" src="https://github.com/user-attachments/assets/d1567ce1-9450-4ce8-8eb6-2d2330fd15a8" />

**Comment by lilailai688** [comment]
Will there be any data leakage problems with this fine-tuning design?

**Comment by iganggang** [comment]
@XUHIT @Maple728 After full fine-tuning, the model achieved a very low loss on the training set, but its loss on the test set was much higher. This suggests possible overfitting. How to address this?
